src/mono/mono/eventpipe/ep-rt-mono.c

```diff
@@ -3172,26 +3174,36 @@ ep_rt_mono_fire_bulk_type_event (BulkTypeEventLogger *type_logger)
 
 	uint32_t values_element_size = 0;
 
-	char *ptr = (char *)type_logger->bulk_type_event_buffer;
+	uint8_t *ptr = type_logger->bulk_type_event_buffer;
 
 	for (uint32_t type_value_index = 0; type_value_index < type_logger->bulk_type_value_count; type_value_index++) {
 		BulkTypeValue *target = &type_logger->bulk_type_values [type_value_index];
 
-		values_element_size += write_event_buffer_int64 (target->fixed_sized_data.type_id, ptr, &ptr);
-		values_element_size += write_event_buffer_int64 (target->fixed_sized_data.module_id, ptr, &ptr);
-		values_element_size += write_event_buffer_int32 (target->fixed_sized_data.type_name_id, ptr, &ptr);
-		values_element_size += write_event_buffer_int32 (target->fixed_sized_data.flags, ptr, &ptr);
-		values_element_size += write_event_buffer_int8 (target->fixed_sized_data.cor_element_type, ptr, &ptr);
+		ep_write_buffer_uint64_t (&ptr, target->fixed_sized_data.type_id);
+		ep_write_buffer_uint64_t (&ptr, target->fixed_sized_data.module_id);
+		ep_write_buffer_uint32_t (&ptr, target->fixed_sized_data.type_name_id);
+		ep_write_buffer_uint32_t (&ptr, target->fixed_sized_data.flags);
+		ep_write_buffer_uint8_t (&ptr, target->fixed_sized_data.cor_element_type);
 
 		g_assert (target->name == NULL);
-		values_element_size += write_event_buffer_int16 (0, ptr, &ptr);
+		size_t num_bytes_utf16_str = ep_write_buffer_string_utf8_to_utf16_t (&ptr, "", 0);
 
-		values_element_size += write_event_buffer_int32 (target->type_parameters_count, ptr, &ptr);
+		ep_write_buffer_uint32_t (&ptr, target->type_parameters_count);
 
 		for (uint32_t i = 0; i < target->type_parameters_count; i++) {
 			uint64_t type_parameter = get_typeid_for_type (target->mono_type_parameters [i]);
-			values_element_size += write_event_buffer_int64 ((int64_t)type_parameter, ptr, &ptr);
+			ep_write_buffer_uint64_t (&ptr, type_parameter);
 		}
+
+		values_element_size += (uint32_t)(
+			sizeof (target->fixed_sized_data.type_id)
+			+ sizeof (target->fixed_sized_data.module_id)
+			+ sizeof (target->fixed_sized_data.type_name_id)
+			+ sizeof (target->fixed_sized_data.flags)
+			+ sizeof (target->fixed_sized_data.cor_element_type)
+			+ num_bytes_utf16_str
+			+ sizeof (target->type_parameters_count)
+			+ (target->type_parameters_count * 8));
```

Comment:
nit: sizeof (uint64_t)

Reply:
or even better, do change as outlined below, have  `ep_write_buffer_` return written byte size.

Reply:
I very much like the idea and changed the `ep_writer_buffer_` functions to return the written number of bytes.